modules/systemdstates/systemdstates.go

```diff
@@ -0,0 +1,76 @@
+package systemdstates
+
+import (
+	"github.com/netdata/go-orchestrator/module"
+	"github.com/netdata/go.d.plugin/pkg/matcher"
+)
+
+type Config struct {
+	Selector matcher.SimpleExpr `yaml:"selector"`
+}
+
+func init() {
+	creator := module.Creator{
+		Defaults: module.Defaults{
+			Disabled:    false,
+			UpdateEvery: 1,
+		},
+		Create: func() module.Module { return New() },
+	}
+
+	module.Register("systemdstates", creator)
+}
+
+// New creates SystemdStates with default values
+func New() *SystemdStates {
+	return &SystemdStates{
+		metrics: make(map[string]int64),
+	}
+}
+
+// SystemdStates SystemdStates module
+type SystemdStates struct {
+	module.Base  // should be embedded by every module
+	Config       `yaml:",inline"`
+	metrics      map[string]int64
+	unitsMatcher matcher.Matcher
+}
+
+// Cleanup makes cleanup
+func (SystemdStates) Cleanup() {}
+
+// Init makes initialization
+func (s *SystemdStates) Init() bool {
+
+	if !s.Selector.Empty() {
+		m, err := s.Selector.Parse()
+		if err != nil {
+			s.Errorf("error on creating per user stats matcher : %v", err)
+		}
+		s.unitsMatcher = matcher.WithCache(m)
+	}
+	return true
+}
+
+// Check makes check
+func (SystemdStates) Check() bool {
+	return true
+}
+
+// Charts creates Charts
+func (s *SystemdStates) Charts() *Charts {
+	return s.charts()
+}
```

Comment:
we need to add `charts` fields to the main structure and return it here

that is what i mean:

```
func New() *SystemdStates {
	return &SystemdStates{
		metrics: make(map[string]int64),
        charts: charts.Copy(),
	}
}

...

type SystemdStates struct {
	module.Base  // should be embedded by every module
	Config       `yaml:",inline"`
	metrics      map[string]int64
	unitsMatcher matcher.Matcher
    charts       *module.Charts
}

func (s *SystemdStates) Charts() *Charts {
	return s.charts
}
```

Reply:
Reason: this is always the pattern if you need to change charts (add/remove charts, add/remove dimensions) in runtime, you add charts instance as an private field to the main structure to get access to it later.

`Charts()` must return instance specific charts, so if there are any changes plugin can detect it and resend charts definition to netadta daemon.